src/migration/1714633817175-SquadTotalPostsTrigger.ts

```diff
@@ -0,0 +1,59 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalPostsTrigger1714633817175 implements MigrationInterface {
+  name = 'SquadTotalPostsTrigger1714633817175';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER), 0) + 1))
+          WHERE   id = NEW."sourceId"
+          AND     type = 'squad';
+          RETURN NEW;
+        END;
+        $$
+      `);
+    queryRunner.query(
+      `CREATE TRIGGER increment_squad_posts_count AFTER INSERT ON "post" FOR EACH ROW EXECUTE PROCEDURE increment_squad_posts_count()`,
+    );
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION decrement_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          IF OLD.deleted = false AND NEW.deleted = true THEN
+            UPDATE  source
+            SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(CAST(flags->>'totalPosts' AS INTEGER) - 1))
```

Comment:
```suggestion
            SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(GREATEST(0, COALESCE(CAST(flags->>'totalPosts' AS INTEGER), 0) - 1)))
```

Just to make sure it handles no value and not go into negative with `GREATEST`.

Reply:
Yep. Makes sense to catch the edge-case.